spec.html

```diff
@@ -155,4 +155,209 @@ <h1>Array.prototype.with ( _index_, _value_ )</h1>
+            <h1>Properties of the %TypedArray% Prototype Object</h1>
+
+            <emu-clause id="sec-%typedarray%.prototype.popped">
+                <h1>%TypedArray%.prototype.popped ( )</h1>
+
+                <p>When the *popped* method is called, the following steps are taken:</p>
+
+                <emu-alg>
+                    1. Let _O_ be the *this* value.
+                    1. Perform ? ValidateTypedArray(_O_).
+                    1. Let _len_ be _O_.[[ArrayLength]].
+                    1. Let _newLen_ be max(_len_ - 1, 0).
+                    1. Let _A_ be ? TypedArraySpeciesCreate(_O_, &laquo; 𝔽(_newLen_) &raquo;).
+                    1. Let _k_ be 0.
+                    1. Repeat, while _k_ &lt; _newLen_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _kValue_ be ! Get(_O_, _Pk_).
+                        1. Perform ? Set(_A_, _Pk_, _kValue_, *true*).
+                        1. Set _k_ to _k_ + 1.
+                    1. Return _A_.
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-%typedarray%.prototype.pushed">
+                <h1>%TypedArray%.prototype.pushed ( ..._items_ )</h1>
+
+                <p>When the *pushed* method is called, the following steps are taken:</p>
+
+                <emu-alg>
+                    1. Let _itemCount_ be the number of elements in _items_.
+                    1. Let _O_ be the *this* value.
+                    1. Perform ? ValidateTypedArray(_O_).
+                    1. Let _len_ be _O_.[[ArrayLength]].
+                    1. Let _newLen_ be _len_ + _itemCount_.
+                    1. Let _A_ be ? TypedArraySpeciesCreate(_O_, &laquo; 𝔽(_newLen_) &raquo;).
+                    1. Let _k_ be 0.
+                    1. Repeat, while _k_ &lt; _len_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _kValue_ be ! Get(_O_, _Pk_).
+                        1. Perform ? Set(_A_, _Pk_, _kValue_, *true*).
+                        1. Set _k_ to _k_ + 1.
+                    1. For each element _E_ of _items_, do
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Perform ? Set(_A_, _Pk_, _E_, *true*).
+                        1. Set _k_ to _k_ + 1.
+                    1. Return _A_.
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-%typedarray%.prototype.reversed">
+                <h1>%TypedArray%.prototype.reversed ( )</h1>
+
+                <p>When the *reversed* method is called, the following steps are taken:</p>
+
+                <emu-alg>
+                    1. Let _O_ be the *this* value.
+                    1. Perform ? ValidateTypedArray(_O_).
+                    1. Let _A_ be ? Call(%TypedArray.prototype.slice%, _O_, &laquo;&raquo;).
+                    1. Perform ? Call(%TypedArray.prototype.reverse%, _A_, &laquo;&raquo;).

```

Comment:
still not convinced about this approach vs a direct algorithm, but this is fine for now ofc

Reply:
Agreed longer term we probably don't want to delegate to the mutator functions like this. Will address as part of #7

Reply:
I've update `reversed` to not call out to the mutator. Just leaving `sorted` as a future TODO.